ff-macros/src/montgomery/mul.rs

```diff
@@ -95,7 +96,11 @@ pub(super) fn mul_assign_impl(
         body.extend(quote! {
             (a.0).0 = scratch[#num_limbs..].try_into().unwrap();
         });
+        if modulus_limbs.last().unwrap() >> 63 != 0 {
```

Comment:
We should pass and use `modulus_has_spare_bit` here.

Reply:
Let me flag this one as "still a todo".

Reply:
Nay, I looked into this, and I think this is the best solution given that this function only uses highly "basic" types and does not involve the MontConfig.

Reply:
ah I mean from the main macro function, where we already calculate `modulus_has_spare_bit`

Reply:
do you mean passing `modulus_has_spare_bit` to this function as well? You can make the change (since you are already here), or if you prefer, I can do the change.

Reply:
Yeah. I can make the change, give me a second.